Index: libibverbs/include/infiniband/verbs.h
===================================================================
--- libibverbs.orig/include/infiniband/verbs.h	2010-09-05 11:35:57.759463753 +0300
+++ libibverbs/include/infiniband/verbs.h	2010-09-05 11:36:27.711461345 +0300
@@ -210,7 +210,8 @@ enum ibv_event_type {
 	IBV_EVENT_SRQ_ERR,
 	IBV_EVENT_SRQ_LIMIT_REACHED,
 	IBV_EVENT_QP_LAST_WQE_REACHED,
-	IBV_EVENT_CLIENT_REREGISTER
+	IBV_EVENT_CLIENT_REREGISTER,
+	IBV_EVENT_GID_CHANGE,
 };
 
 enum ibv_event_flags {
Index: libibverbs/examples/asyncwatch.c
===================================================================
--- libibverbs.orig/examples/asyncwatch.c	2010-09-05 11:45:27.955963185 +0300
+++ libibverbs/examples/asyncwatch.c	2010-09-05 11:52:27.037843960 +0300
@@ -57,6 +57,8 @@ static const char *event_name_str(enum i
 		return "IBV_EVENT_SM_CHANGE";
 	case IBV_EVENT_CLIENT_REREGISTER:
 		return "IBV_EVENT_CLIENT_REREGISTER";
+	case IBV_EVENT_GID_CHANGE:
+		return "IBV_EVENT_GID_CHANGE";
 
 	case IBV_EVENT_CQ_ERR:
 	case IBV_EVENT_QP_FATAL:
